polynote-frontend/polynote/ui.js

```diff
@@ -1384,40 +1406,47 @@ export class MainUI extends EventTarget {
            this.socket.send(new messages.CancelTasks(this.currentNotebookPath));
         });
 
+        this.toolbarUI.addEventListener('Undo', () => {
```

Comment:
Note: the motivation for this PR was because I'm thinking to maybe use the edit buffer to also be an undo buffer of sorts. So that's why there's sprinkling of Undo stuff in here. Doesn't harm anything now, and will complete it in follow-up PR, but thought this might as well get its own PR since it's a good bugfix.